library/L1_Peripheral/lpc40xx/test/capture_test.cpp

```diff
@@ -0,0 +1,153 @@
+    Mock<sjsu::SystemController> mock_system_controller;
+    Fake(Method(mock_system_controller, PowerUpPeripheral));
+
+    constexpr units::frequency::hertz_t kTestSystemFrequency = 4_MHz;
+    constexpr int kTestPeripheralClockDivider = 1;
+
+    When(Method(mock_system_controller, GetSystemFrequency))
+        .AlwaysReturn(kTestSystemFrequency);
+    When(Method(mock_system_controller, GetPeripheralClockDivider))
+        .AlwaysReturn(kTestPeripheralClockDivider);
+
+    Mock<sjsu::InterruptController> mock_interrupt_controller;
+    Fake(Method(mock_interrupt_controller, Register));
+
+    Capture test_subject(kTestTimer,
+        mock_system_controller.get(),
+        mock_interrupt_controller.get());
+
+    SECTION("Capture Initialize")
+    {
+        constexpr units::frequency::hertz_t kTestFrequency = 1_MHz;
+
+        test_subject.Initialize(kTestFrequency, &DummyCallback);
+        int prescaler = (kTestSystemFrequency.to<int32_t>() /
+                kTestPeripheralClockDivider) / kTestFrequency.to<int32_t>();
+
+        Verify(Method(mock_interrupt_controller, Register)
+        .Matching(
+            [kTestTimer](
+                sjsu::InterruptController::RegistrationInfo_t info) {
+              return (info.interrupt_request_number ==
+                      kTestTimer.channel.irq) &&
+                    (info.interrupt_service_routine ==
+                      kTestTimer.isr) &&
+                    (info.enable_interrupt == true) &&
+                    (info.priority == -1);
+            }));
+
+        CHECK(test_timer_register.PR == prescaler);
+        CHECK(bit::Read(test_timer_register.TCR, 0) == true);
+    }  // end section initialize
+
+    SECTION("Capture Config")
+    {
+      for (size_t channel = 0; channel < std::size(kChannelList); channel++)
+      {
+        int position = 0;
+        if (channel)
+        {
+          position = 3;
+        }
+
+        for (size_t mode = 0; mode < std::size(kModeList); mode++)
+        {
+          test_subject.ConfigureCapture(kChannelList[channel], kModeList[mode]);
+          CHECK(bit::Extract(test_timer_register.CCR, position, 2) ==
+                kModeList[mode]);
+        }
+        }
+    }  // end section capture config
+
+    SECTION("Capture Interrupt")
+    {
+      for (size_t channel = 0; channel < std::size(kChannelList); channel++)
+      {
+        int position = 2;
+        if (channel)
+        {
+            position = 5;
+        }
+
+        test_subject.EnableCaptureInterrupt(kChannelList[channel], true);
+        CHECK(bit::Read(test_timer_register.CCR, position) == true);
+        test_subject.EnableCaptureInterrupt(kChannelList[channel], false);
+        CHECK(bit::Read(test_timer_register.CCR, position) == false);
+        }
+    }  // end section capture interrupt
+
+    SECTION("Capture Interrupt Handler")
+    {
+      test_timer_register.IR = 0;
+      kTestTimer.isr();
+      CHECK(bit::Extract(test_timer_register.IR, 4, 2) == 0b11);
```

Comment:
Missing a test for a call to DummyCallback, make sure that your ISR does in fact call it. It may be helpful to use the FFF library to mock the calllback and get invocation history.